test/attribute/attribute_read_io_test.cpp

```diff
@@ -0,0 +1,150 @@
+//
+// (c) Copyright 2017 DESY,ESS
+//
+// This file is part of h5cpp.
+//
+// This library is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation; either version 2.1 of the License, or
+// (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty ofMERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+// License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this library; if not, write to the
+// Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+// Boston, MA  02110-1301 USA
+// ===========================================================================
+//
+// Author: Jan Kotanski <jan.kotanski@desy.de>
+// Created on: Jun 19, 2018
+//
+#include <gtest/gtest.h>
+#include <h5cpp/hdf5.hpp>
+#include <string>
+
+using namespace hdf5;
+
+class AttributeReadIO: public testing::Test
+{};
+
+TEST_F(AttributeReadIO, read_string_scalar_20)
+{
+
+  auto file = file::create("AttributeReadIO.h5",
+			   file::AccessFlags::TRUNCATE);
+  auto r = file.root();
+  std::string data = "hello";
+
+  datatype::String fixed_type = datatype::String::fixed(20);
+
+  auto a = r.attributes.create("scalar20", fixed_type, dataspace::Scalar());
+  a.write(data, fixed_type);
+  std::string buffer;
+  a.read(buffer); // causes terminate called after throwing an instance of 'std::bad_alloc'
```

Comment:
If an exception thrown is the expected (desired behavior), then you can test it with `EXPECT_THROW`. Also see other tests.